app/src/main/java/org/schabi/newpipe/local/bookmark/BookmarkFragment.java

```diff
@@ -247,12 +314,242 @@ protected void resetFragment() {
+                .subscribe(longs -> { /*Do nothing on success*/ }, throwable -> showError(
+                        new ErrorInfo(throwable,
+                                UserAction.REQUESTED_BOOKMARK,
+                                "Changing playlist name")));
+        disposables.add(disposable);
+    }
+
+    private void deleteItem(final PlaylistLocalItem item) {
+        if (itemListAdapter == null) {
+            return;
+        }
+        itemListAdapter.removeItem(item);
+
+        debounceSaver.saveChanges();
+    }
+
+    private void checkDisplayIndexModified(@NonNull final List<PlaylistLocalItem> result) {
+        if (debounceSaver != null && debounceSaver.getIsModified()) {
+            return;
+        }
+
+        displayIndexInDatabase.clear();
+
+        // If the display index does not match actual index in the list, update the display index.
+        // This may happen when a new list is created
+        // or on the first run after database migration
+        // or display index is not continuous for some reason
+        // or the user changes the display index.
+        boolean isDisplayIndexModified = false;
+        for (int i = 0; i < result.size(); i++) {
+            final PlaylistLocalItem item = result.get(i);
+            if (item.getDisplayIndex() != i) {
+                isDisplayIndexModified = true;
+            }
+
+            // Updating display index in the item does not affect the value inserts into
+            // database, which will be recalculated during the database update. Updating
+            // display index in the item here is to determine whether it is recently modified.
+            // Save the index read from the database.
+            if (item instanceof PlaylistMetadataEntry) {
+
+                displayIndexInDatabase.put(new Pair<>(((PlaylistMetadataEntry) item).uid,
+                        LocalItem.LocalItemType.PLAYLIST_LOCAL_ITEM), item.getDisplayIndex());
+                ((PlaylistMetadataEntry) item).displayIndex = i;
+
+            } else if (item instanceof PlaylistRemoteEntity) {
+
+                displayIndexInDatabase.put(new Pair<>(((PlaylistRemoteEntity) item).getUid(),
+                                LocalItem.LocalItemType.PLAYLIST_REMOTE_ITEM),
+                        item.getDisplayIndex());
+                ((PlaylistRemoteEntity) item).setDisplayIndex(i);
+
+            }
+        }
+
+        if (debounceSaver != null && isDisplayIndexModified) {
+            debounceSaver.saveChanges();
+        }
+    }
+
+    @Override
+    public void saveImmediate() {
+        if (itemListAdapter == null) {
+            return;
+        }
+
+        // List must be loaded and modified in order to save
+        if (isLoadingComplete == null || debounceSaver == null
+                || !isLoadingComplete.get() || !debounceSaver.getIsModified()) {
+            Log.w(TAG, "Attempting to save playlists in bookmark when bookmark "
```

Comment:
Since this is always called in `onPause`, I would not print a warning here

Reply:
Nope. It will also be called 10 seconds after the last change, not only in `onPause`.

https://github.com/GGAutomaton/NewPipe/blob/6526ff1612ad430325a5bf8af7b425055306a0e6/app/src/main/java/org/schabi/newpipe/util/DebounceSaver.java#L51-L58

Reply:
Yes, I know. But I mean, this is, among other places, always called in `onPause`. So the warning will be generated each time this method is called from `onPause`, which is not good...

Reply:
Deleted.